Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py

```diff
@@ -850,6 +860,158 @@ def set_blob_properties_command(client: Client, args: Dict[str, Any]) -> Command
+        sig = base64.b64encode(signed_hmac_sha256.digest())
+        token = {
+            'sp': sp,
+            'st': signedstart,
+            'se': expiry,
+            'spr': "https",
+            'sv': "2021-06-08",
+            'sr': sr,
+            'sig': sig
+        }
+
+    sas_token = urllib.parse.urlencode(token)
+    return sas_token
+
+
+def check_valid_permission(string, per):
+    """
+    Check the permissions follows valid permission order.
+
+    Args:
+        string : valid permissions order
+        per : permissions given
+
+    Returns:
+        bool
+
+    """
+    l = len(per)
+    if len(string) < l:
+        return False
+    for i in range(l - 1):
+        x = per[i]
+        y = per[i + 1]
+        last = string.rindex(x)
+        first = string.index(y)
+        if last == -1 or first == -1 or last > first:
+            return False
+    return True
+
+
+def generate_sas_token(client: Client, args: dict) -> CommandResults:
+    """
+    Generate sas url for Container.
+
+    Args:
+        client (Client): Azure Blob Storage API client.
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs and raw response for XSOAR.
+
+    """
+    container_name = args.get("container_name")
+    signedResource = args.get("signed_resources")
+    signedPermissions = args.get("signed_permissions")
+    valid_permissions = "racwdxltmeop"
+    signedIp = args.get("signed_ip")
+    # Check Permissions
+    if check_valid_permission(valid_permissions, signedPermissions):
+        # Set start time
+        signedstart = str((datetime.utcnow() - timedelta(minutes=2)).strftime("%Y-%m-%dT%H:%M:%SZ"))
+        account_key = demisto.params().get("key")
+        time_taken = int(args.get('expiry_time'))
+        signedExpiry = str((datetime.utcnow() + timedelta(hours=time_taken)).strftime("%Y-%m-%dT%H:%M:%SZ"))
+        url_suffix = f"{container_name}"
+        canonicalizedResource = f"/blob/{storage_account_name}/{container_name}"
+        url = client.get_base_url() + url_suffix
+        uri = urllib.parse.quote_plus(f"{url}")
+        sas_token = generate_sig(account_key, canonicalizedResource, signedPermissions,
+                                 signedstart, signedExpiry, signedResource, signedIp)
+        sas_url = f"{url}?{sas_token}"
+        if sas_url:
+            res_data = sas_url
+            markdown = tableToMarkdown('Azure storage container SAS url', res_data, headers=[container_name])
+
+            result = CommandResults(
+                readable_output=markdown,
+                outputs_prefix='AzureStorageContainer.Container',
+                outputs_key_field=container_name,
+                outputs=res_data
+            )
+            return result
+        else:
+            return_error(f"Error")
+    else:
+        return_error(f"Permission are invalid or in wrong order.")
```

Comment:
Can you please add an example i.e. Permission are invalid or in wrong order, they should be in the following structure <example>.

Reply:
I added a correct order for permission like below 
Permission are invalid or in wrong order. Correct order for permissions are racwdl